Expand DocC: config-type reference, rebalances/events, error handling, observability (#248)
swift-server/swift-kafka-client: Sources/Kafka/Kafka.docc/Observability.md

> # Observability

Reviewer: All the sibling articles are gerunds (Producing... Consuming... etc) - so what do you think about:
```suggestion
# Observing  Kafka clients
```

(observing may be too overloaded in the context of queues here, so this is a weak suggestion - Observability is pretty well understood - more it just visually stood out as awkward when I scanned it with it's neighbors)

Emit metrics and structured logs from the producer and consumer.

## Overview

The Kafka client integrates with [swift-metrics](https://github.com/apple/swift-metrics) for runtime metrics and [swift-log](https://github.com/apple/swift-log) for structured logging. Both are opt-in through the client configuration and the `Logger` you provide.

## Emit metrics

The client periodically samples internal statistics and records them into [swift-metrics](https://github.com/apple/swift-metrics) gauges that you supply. Configure this through the ``KafkaConsumerConfig/metrics`` (or ``KafkaProducerConfig/metrics``) property: set an update interval and assign a `Gauge` to each statistic you want to track.

> 

Reviewer: ```suggestion

The following example illustrates configuring a Kafka client with metrics:

```

> The client periodically samples librdkafka's internal statistics and records them into [swift-metrics](https://github.com/apple/swift-metrics) gauges that you supply. Configure this through the ``KafkaConsumerConfig/metrics`` (or ``KafkaProducerConfig/metrics``) property: set an update interval and assign a `Gauge` to each statistic you want to track.

Reviewer: ```suggestion
The client periodically samples internal statistics and records them into [swift-metrics](https://github.com/apple/swift-metrics) gauges that you supply. Configure this through the ``KafkaConsumerConfig/metrics`` (or ``KafkaProducerConfig/metrics``) property: set an update interval and assign a `Gauge` to each statistic you want to track.
```

Is it OK if we drop the specifics of librdkafka here - or is it meaningful that it's specifically that library? It adds in a heavy jargon bit here that's technically accurate, but kind of reads like an exposed implementation detail that doesn't benefit the reader to me.

```swift
import Kafka
import Metrics

var config = KafkaConsumerConfig()
config.bootstrapServers = ["localhost:9092"]
config.consumptionStrategy = .group(id: "example-group", topics: ["topic-name"])

// Sample statistics once per second into swift-metrics gauges.
config.metrics.updateInterval = .seconds(1)
config.metrics.totalKafkaBrokerRequests = Gauge(label: "kafka_consumer_broker_requests")
config.metrics.totalKafkaBrokerMessagesReceived = Gauge(label: "kafka_consumer_messages_received")
config.metrics.queuedOperation = Gauge(label: "kafka_consumer_queued_operations")

let consumer = try KafkaConsumer(config: config, logger: logger)
```

> Metrics are emitted only when ``KafkaConfiguration/ConsumerMetrics/updateInterval`` is set **and** at least one gauge is assigned; otherwise the client skips statistics collection entirely. The producer exposes the same pattern through ``KafkaProducerConfig/metrics`` with producer-specific gauges such as ``KafkaConfiguration/ProducerMetrics/queuedProducerMessages`` and ``KafkaConfiguration/ProducerMetrics/totalKafkaBrokerMessagesSent``.

Reviewer: ```suggestion
The client emits metrics only when ``KafkaConfiguration/ConsumerMetrics/updateInterval`` is set **and** at least one gauge is assigned; otherwise the client skips statistics collection entirely. The producer exposes the same pattern through ``KafkaProducerConfig/metrics`` with producer-specific gauges such as ``KafkaConfiguration/ProducerMetrics/queuedProducerMessages`` and ``KafkaConfiguration/ProducerMetrics/totalKafkaBrokerMessagesSent``.
```

(passive voice -> active voice)

The gauges you assign are ordinary [swift-metrics](https://github.com/apple/swift-metrics) types, so the values reach whatever metrics backend you bootstrap through `MetricsSystem`.

> ## Structured logging

Reviewer: ```suggestion
## Emit structured logs
```
(this keeps the headings parallel - both imperative vs. noun phrases)
